{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n",
      "Exception in Tkinter callback\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.5/tkinter/__init__.py\", line 1562, in __call__\n",
      "    return self.func(*args)\n",
      "  File \"<ipython-input-4-132339a7be0b>\", line 8, in newN\n",
      "    show(cnv, N, P)\n",
      "NameError: name 'cnv' is not defined\n"
     ]
    }
   ],
   "source": [
    "from tkinter import *\n",
    "from math import *\n",
    "\n",
    "\n",
    "def newN(value):\n",
    "    global N\n",
    "    N=int(value)\n",
    "    show(cnv, N, P)\n",
    "\n",
    "def newP(value):\n",
    "    global P\n",
    "    P=int(value)\n",
    "    show(cnv, N, P)\n",
    "\n",
    "def show(cnv, n, p):\n",
    "    cnv.delete(\"all\")\n",
    "    t=2*pi/p\n",
    "    base=[(R*cos(k*t), R*sin(k*t)) for k in range(p)]\n",
    "    draw_base(cnv, base, a, b)\n",
    "\n",
    "\n",
    "def passage(x,y, a, b):\n",
    "    return (a+x, b-y)\n",
    "\n",
    "def produit(k, n, p):\n",
    "    return k*n % p\n",
    "\n",
    "def segment(cnv, k, n, p, base, a, b):\n",
    "    # base : les coordonnées des points du cercle\n",
    "    # on trace le produit prod=k*n\n",
    "    # k -> A\n",
    "    # prod -> B\n",
    "    prod=produit(k, n, p)\n",
    "    xA, yA=base[k]\n",
    "    xB, yB=base[prod]    \n",
    "    A=passage(xA, yA, a, b)\n",
    "    B=passage(xB, yB, a, b)\n",
    "    cnv.create_line(A, B)\n",
    "    \n",
    "def dot(cnv, C, R=6, color='red'):\n",
    "    xC, yC=C\n",
    "    return cnv.create_oval(xC-R,yC-R,xC+R, yC+R, fill=color, outline=color)\n",
    "  \n",
    "def draw_base(cnv, p, base, a, b):\n",
    "    # Les sommets sur le cercle\n",
    "    for k in range(p):\n",
    "        x, y=base[k]\n",
    "        X, Y=passage(x, y, a, b)\n",
    "        dot(cnv, (X, Y), R=2, color='black')\n",
    "    \n",
    "def draw(n, p, R):\n",
    "    master=Tk()\n",
    "    a=b=1.2*R\n",
    "    cnv=Canvas(master, width=2*a, height=2*b, bg='ivory')\n",
    "    cnv.pack()\n",
    "    # Les points du cercle\n",
    "    base=[(R*cos(2*k*pi/p), R*sin(2*k*pi/p)) for k in range(p)]\n",
    "    draw_base(cnv, p, base, a, b)\n",
    "    # Tracé de la table\n",
    "    for k in range(p):\n",
    "        segment(cnv, k, n, p, base, a, b)    \n",
    "  \n",
    "    NbrDepts = Scale(master,label=\"Nombre de points sur le cercle\",font=\"Arial 15 bold\", orient = \"horizontal\", command= newN, from_=2, to=500,length=250)\n",
    "    NbrDepts.pack()\n",
    "    Latablede = Scale(master,label=\"Table de :\",font=\"Arial 15 bold\", orient = \"horizontal\", command= newP, from_=2, to=50,length=250)\n",
    "    Latablede.pack()\n",
    "    master.mainloop()\n",
    "  \n",
    "\n",
    "\n",
    "\n",
    " \n",
    "p=100 # nombre de points sur le cercle\n",
    "R=200\n",
    "n=5 # Table de n\n",
    "draw(n, p, R)\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
